vue/components/ui/molecules/select-list-controller/select-list-controller.vue

```diff
@@ -0,0 +1,192 @@
+}
+
+.to-left-all-icon {
+    background: url("~./assets/group-4.svg") no-repeat center;
+}
+
+.content-item {
+    color: #1d2631;
+    cursor: pointer;
+    font-size: 14px;
+    height: 32px;
+    width: 303px;
+}
+
+.content-item > span {
+    line-height: 32px;
+    margin-left: 5px;
+}
+
+.content-item.selected,
+.content-item:hover {
+    background-color: #1d2631;
+    color: #ffffff;
+}
+</style>
+
+<script>
+import Vue from "vue";
+
+export const SelectListController = Vue.component("select-list-controller", {
+    inheritAttrs: false,
+    props: {
+        leftItems: {
+            type: Array,
+            default: () => []
+        },
+        rightItems: {
+            type: Array,
+            default: () => []
+        },
+        valuesLeft: {
+            type: Object,
+            default: () => ({})
+        },
+        valuesRight: {
+            type: Object,
+            default: () => ({})
+        },
+        leftTitle: {
+            type: String,
+            default: null
+        },
+        rightTile: {
+            type: String,
+            default: null
+        }
+    },
+    data: function() {
+        return {
+            selectedItemsLeft: this.valuesLeft,
+            selectedItemsRight: this.valuesRight,
+            leftData: this.leftItems,
+            rightData: this.rightItems
+        };
+    },
+    mounted: async function() {},
+    methods: {
+        removeItem(value, fromLeft = true) {
+            let index = -1;
+            if (fromLeft) {
+                index = this.leftData.findIndex(element => element.value === value);
+                this.leftData.splice(index, 1);
+            }
+            if (!fromLeft) {
+                index = this.rightData.findIndex(element => element.value === value);
+                this.rightData.splice(index, 1);
+            }
+        },
+        moveSelected(toRight = true) {
+            let values = {};
+
+            if (toRight) {
+                const selectedItems = Object.entries(this.selectedItemsLeft);
+                selectedItems.forEach(element => {
+                    values.value = element[0];
+                    this.rightData.push(values);
+                    this.$delete(this.selectedItemsLeft, values.value);
+                    this.removeItem(values.value, true);
+                    values = {};
+                });
+            } else {
+                const selectedItems = Object.entries(this.selectedItemsRight);
+                selectedItems.forEach(element => {
+                    values.value = element[0];
+                    this.leftData.push(values);
+                    this.$delete(this.selectedItemsRight, values.value);
+                    this.removeItem(values.value, false);
+                    values = {};
+                });
+            }
+            this.$emit("update:value", this.rightData);
+        },
+        moveAll(toRight = true) {

```

Comment:
maybe applying the onMoveAll logic, that calls this function. The same for other functions called from the buttons